ErikBagratyan/Hash_Table

```diff
@@ -0,0 +1,83 @@
+#include <iostream>
+#include <vector>
+#include <list>
+#include <limits>
+#include <functional>
+
+std::hash<int> h;
+
+class Hash
+{
+  public:
+    Hash(int _size) 
+    :   table_size(_size)
+    {
+        hash_table.resize(table_size);
+    }
+    
+    void insert(int x)
+    {
+       int index = h(x) % table_size;
+       hash_table[index].push_back(x);
+    }
+    
+    void deleteElem(int x)
+    {
+        int index = h(x) % table_size;
+        auto it = hash_table[index].begin();
+        for(; it != hash_table[index].end(); ++it)
+        {
+            if(*it == x) 
+            {
+                break;
+            }
+        }
+        if(it != hash_table[index].end()) hash_table[index].erase(it);
```

Comment:
The only case this doesn't work is then there are multiple elements of the same value you are trying to remove, but that also depends on how you define the functionality.